### DISCUSSION TITLE
Recipe Ideas 🧑‍🍳

### DISCUSSION THREAD

**Discussion by delucis** [withastro/docs#2894] [Ideas]
### What’s cookin’ docs?

Use this discussion to share ideas for new [“Recipes”](https://docs.astro.build/en/recipes/) that you’d like to see added to Docs. (Or recipes you’d like to write!)

#### What’s a recipe?
Recipes are step-by-step how-to guides for accomplishing a specific task in Astro. [Check out our full guide to writing recipes.](https://contribute.docs.astro.build/guides/recipe-writing/)

#### How many recipes can I suggest?
As many as you want! If possible, use a separate post for each idea. This will let people vote for them one by one.

#### See a suggestion you like?
Upvote it so we know! ⬆️

**Comment by delucis** [comment]
How to make light/dark themes with Astro?

**Reply by xijaja** [reply]
我知道的方法虽然可以实现，但那非常麻烦。

**Reply by hippotastic** [reply]
> 我知道的方法虽然可以实现，但那非常麻烦。

Google Translation for anyone like me who can't read the original comment: "The method I know can be done, but it is very troublesome."
(Which would have been a very fitting reply to the i18n recipe idea as well 😄)

**Reply by tony-sull** [reply]
Oh I have a few good examples of this!  Avoiding a flash on page load is always the trick, I'll see if I can boil it down to a simple recipe example

**Comment by delucis** [comment]
How to build i18n features with Astro? (UI translation, content, language switching and redirects)

**Reply by darbyjack** [reply]
Hey there! I'd love to see some kind of recipe with being able to use translation systems and pulling them into i18n support. While some people can just clone and modify the files manually, there are tools like https://crowdin.com/ that allow users to go in and suggest / approve translations. These end up syncing back to Github via an integration (they support others too) so you can just use those translated files inside of i18n.

**Reply by delucis** [reply]
We shipped an i18n recipe today!

https://docs.astro.build/en/recipes/i18n/

**Reply by teinett** [reply]
Here https://github.com/withastro/docs/pull/2895#issuecomment-1483113174 there is mention about language switcher for localizations: /en/about => /fr/about.

Current language switcher changes the language, but skip the page url: /en/about => /fr. It's not good UX for users.

Please, add info to this recipe about case with the same slug for url: /en/about => /fr/about

**Reply by artursopelnik** [reply]
Thank you for shipping the i18n recipe @delucis. Only a few important functions are currently not documented.

How to switch language between specific pages to get a better UX @teinett and how to hide the default language in the URL.

I did that once. Check out my PR:
https://github.com/withastro/docs/pull/4161

**Reply by sarah11918** [reply]
✅ Completed! [Add i18n feature recipe](https://docs.astro.build/en/recipes/i18n/)

**Comment by hippotastic** [comment]
How to implement a contact form in Astro

**Reply by hippotastic** [reply]
Note that while we do have a forms example here, we are not using the term "contact form" anywhere, which is what I searched for: https://docs.astro.build/en/recipes/build-forms-api/

Also, this example does not actually send e-mails or any other form of notification to the website owner, which is usually what people are looking for in contact forms.

**Reply by sarah11918** [reply]
Forms is a great example!

**Reply by hippotastic** [reply]
Can we find a free service or backend library that is able to turn these form submissions into e-mails to the site owner?

**Comment by hippotastic** [comment]
How to implement search/filtering of structured content in Astro? (There could be one example using SSR, and one that builds a search index on the server and passes it to a framework component that does the searching/filtering live on the client)

**Reply by sarah11918** [reply]
What do you think about something like what we have in docs for the deploy guides, like filtering SSR or SSG deployment options?

**Comment by hippotastic** [comment]
How to load data from MongoDB into Astro

**Comment by dsomel21** [comment]
How to open and close a modal? I did one with Nanostores. I can possibly write one if you review the code and think my approach is good.

**Reply by christophermickle** [reply]
Oh! I can totally do this :D

**Reply by wassfila** [reply]
I did one with custom events, possible too.

**Reply by tony-sull** [reply]
+1 on custom events!  We also might finally be at a point where the `<dialog>` element is actually usable in modern browsers 🤞

**Reply by sarah11918** [reply]
So many peeps offering this one! Remember, we also have "Community Recipes" so if you do write something on your own blog, we can link to it! Make sure to consult our RECIPES.md in the docs repo for guidelines, if you're unsure what kind of articles we link to!

**Reply by teinett** [reply]
Did you try headless UI components? Example: [Dialog from Radix UI](https://www.radix-ui.com/docs/primitives/components/dialog)

**Comment by hippotastic** [comment]
How to allow user comments on my blog posts

**Reply by sarah11918** [reply]
Someone was just asking about this yesterday! I know disqus is popular, but perhaps overkill. I use giscus, but that requires commenters to have a GitHub account. Good for tech blogs, but maybe less accessible in general.

Anyone have any thoughts on good commenting systems to use in a recipe?

**Comment by sasoria** [comment]
How to set up micro-frontends with EcmaScript Modules. I've written an [article](https://medium.com/@sergio.a.soria/setting-up-micro-frontends-with-astro-and-ecma-script-modules-137340d2c520) about it with a code [example](https://github.com/sasoria/astro-microfrontends) on GitHub. Happy to contribute with anything!

**Comment by sasoria** [comment]
How to get dependencies from [esm.sh](https://esm.sh/). I've made an example [here](https://github.com/sasoria/astro-importmaps-rt).

**Comment by christophermickle** [comment]
How to get Firebase and Firestore up and running for an Astro Project.

**Reply by sarah11918** [reply]
I think @kevinzunigacuellar  is working on some examples right now!

**Reply by kevinzunigacuellar** [reply]
Yeah, so far I have authentication set up. Feel free to take a look https://github.com/kevinzunigacuellar/astro-firebase

**Comment by eddiek50** [comment]
How to use Material UI components (Emotion) with Astro

**Reply by delucis** [reply]
This one could be tricky at the moment. I don't believe we support Emotion currently: https://github.com/withastro/astro/issues/4432

**Comment by AbhiPrasad** [comment]
Using Astro with [Sentry](https://sentry.io/) to monitor your errors. We at Sentry have been exploring an [official Sentry plugin](https://github.com/getsentry/sentry-javascript/discussions/7378), but in the meantime having some kind of preliminary docs might be a good step!

Happy to help write something up!

**Reply by sarah11918** [reply]
Ooh, I don't think we have anything like this! Would absolutely be interested in seeing something for this!

**Reply by sarah11918** [reply]
✅ We now have [a guide page for Sentry](https://docs.astro.build/en/guides/backend/sentry/)! :partying_face:

**Comment by Lucaskolinger** [comment]
How to integrate Supabase with Astro? Maybe also with Authentication.

**Reply by sarah11918** [reply]
I know @kevinzunigacuellar  is working on a Firebase example project right now. Would be interesting to see the same thing replicated in a Supabase example, too!

**Reply by Lucaskolinger** [reply]
https://www.aleksandra.codes/astro-supabase

This is the most useful tutorial I could find. I don't know if it's suitable as a community recipe but maybe it could be considered.

**Reply by sarah11918** [reply]
✅ We now have [a Supabase guide page](https://docs.astro.build/en/guides/backend/supabase/) with the above tutorial linked as a community recipe. :partying_face:

**Comment by sarah11918** [comment]
- how to work with cookies and cookies notice for GDPR and CCPA laws. How to load partytown 3rd party scripts only when cookies are accepted.

**Comment by sarah11918** [comment]
- how to use i18n for content consumed from API

**Comment by sarah11918** [comment]
- how to make manual language switch and automatic application preferred language from the browser settings.

**Reply by teinett** [reply]
I did this for /src/pages/index.astro
```
---
const headers = Astro.request.headers;
const langHeaders = headers.get('accept-language')

if (langHeaders === null || langHeaders === undefined || langHeaders === '') {
    return Astro.redirect('/en');
} else {
    const lang = langHeaders.split(',')[0].split('-')[0];
    if (lang !== 'es' && lang !== 'ru') {
        return Astro.redirect('/en');
    } else {
        return Astro.redirect(`/${lang}`);
    }
}
---


```

**Comment by sarah11918** [comment]
- how to integrate Astro and StrapiJS CMS

**Reply by sarah11918** [reply]
✅ We now have [a Strapi & Astro CMS guide](https://docs.astro.build/en/guides/cms/strapi/)! 🥳

**Comment by sarah11918** [comment]
- how to make a special look for blocks inside MD/MDX, for example, block «tips» in Astro Docs

**Reply by kevinzunigacuellar** [reply]
maybe plugin recommendations? 👀

**Reply by teinett** [reply]
Answer: use markdown transformation and markdown directives. 

Install packages:`npm i remark-directive unified unist-util-remove unist-util-visit`

Add transformation: `/src/utils/useMarkdownNotes.js`
```
import { visit } from "unist-util-visit"

// This plugin is an example to turn `:::note ` into aside blocks.
// Variants of directive: note, info, tip, warning, caution

/** @type {import('unified').Plugin<[], import('mdast').Root>} */
export default function markdownAside() {
  return (tree) => {
    visit(tree, (node) => {
      if ( node.type === "containerDirective" ) {
        if (!["note", "info", "tip", "warning", "caution"].includes(node.name)) return

        const data = node.data || (node.data = {})
        const attributes = node.attributes || {}
        const type = node.name

        data.hName = 'aside'
        data.hProperties = {
            dataType: type
        }
      }
    })
  }
}
```

Using in markdown: `src/content/posts/12345.md`
```
# Moscow Domodedovo Airport FAQs

::TOC

👋 Hi, I love emojis a lot 🤓[^1]

[^1]: My reference.

**How many airports are there in Salta?**

:::note
There is no separate MDX configuration for [including pages marked as draft in the build](https://docs.astro.build/en/reference/configuration-reference/#markdowndrafts). This Markdown setting will be respected by both Markdown and MDX files and cannot be overriden for MDX files specifically.
:::

There is only <mark>very important words</mark> one airport in Salta, Martin Miguel de Guemes International Airport.

:::warning[Heading for warning]
MDX does not support passing remark and rehype plugins as a string.

You should install, import, and apply the plugin function instead.
:::

Some sentence after directive.
```

Markdown React component: `src/components/MarkdownText.js`

```
import styles from './MarkdownText.module.css';
import ArticleAside from '@/components/ArticleAside'
import ReactMarkdown from "react-markdown";
import directive from "remark-directive";
import markdownAside from "@/utils/useMarkdownNotes"

const Markdown = ({ markdown, customComponents }) => {

    const components = {
        // replace aside html tag with ArticleAside component
        aside: ArticleAside,
    };

    return (
            <div itemProp="articleBody" className='prose'>
            <ReactMarkdown
                components={{ ...components, ...customComponents }}

                remarkPlugins={[
                    directive,
                    markdownAside
                ]}

                >
                {markdown}
            </ReactMarkdown>
        </div>
    );
}

export default Markdown;
```

**Comment by sarah11918** [comment]
- how to work with iframe blocks that add some js files for its work.

**Comment by sarah11918** [comment]
- how to make the breadcrumbs component for the current page if there is only the URL presented.

**Comment by sarah11918** [comment]
- how to make monorepo with Astro components library and Astro project

**Comment by sarah11918** [comment]
- how to work with maps, for example, Leaflet and OpenMaps.

**Reply by teinett** [reply]
Some helpful info

- **[`react-leaflet`](https://react-leaflet.js.org/)**: This library provides `React` components for building interactive maps using the `Leaflet` JavaScript library.
- **[`react-leaflet-draw`](https://www.npmjs.com/package/react-leaflet-draw)**: This library extends `react-leaflet` by providing components for adding drawing functionalities to the map.

Example of realization in React, but not in AstroJS: https://strapi.io/blog/how-to-build-an-interactive-map-with-react-leaflet-and-strapi

**Comment by sarah11918** [comment]
- how to work with lazyYoutube scripts and other lazy-version for embeds of social media posts.

**Comment by sarah11918** [comment]
- How to add an icon for external links in Markdown/MDX files like in Wikipedia.

**Reply by sarah11918** [reply]
Suggestion from Kapa AI:

You can add icons to external links in your Markdown files using a rehype plugin called `rehype-external-links`. Here's how you can do it:

1. Install the `rehype-external-links` plugin. You can install it using npm, pnpm, or Yarn:

```bash
npm install rehype-external-links
```
```bash
pnpm add rehype-external-links
```
```bash
yarn add rehype-external-links
```
[source](https://docs.astro.build/en/recipes/external-links/#_top)

2. Import the plugin into your `astro.config.mjs` file. Pass `rehypeExternalLinks` to the `rehypePlugins` array, along with an options object that includes a content property. Set this property’s `type` to `text` if you want to add plain text to the end of the link. To add HTML to the end of the link instead, set the property `type` to `raw`.

```javascript
// ...import rehypeExternalLinks from 'rehype-external-links';
export default defineConfig({  // ...  markdown: {    rehypePlugins: [      [        rehypeExternalLinks,        {          content: { type: 'text', value: ' 🔗' }        }      ],    ]  },});
```
[source](https://docs.astro.build/en/recipes/external-links/#_top)

Please note that the value of the `content` property is not represented in the accessibility tree. As such, it’s best to make clear that the link is external in the surrounding content, rather than relying on the icon alone. [source](https://docs.astro.build/en/recipes/external-links/#_top)

**Comment by sarah11918** [comment]
- how to make browser push subscriptions on the Astro website

**Comment by sarah11918** [comment]
How to include CodeSandbox's Sandpack embed (in `.astro`, `.mdx`)

I've had to answer the same question twice within a week, and go hunting up old repository links showing what I did back in Astro v0.21 back when you could do this directly in (Astro-flavoured) Markdown.

> I had this working fine in an older version of Astro! (Back when you could have components directly in Markdown files, so be advised that is different now.)
>
> Since I haven't converted my blog posts to MDX yet, I just didn't port over that particular blog post. But I can't see any reason why Sandpack shouldn't work in MDX, and it certainly does (did) in an .astro file. 

> In case this helps, here's the file of my old blog post about it: https://github.com/sarah11918/sastro-v21/blob/main/src/pages/posts/react-sandpack-code-playground-in-astro-markdown.md
> 
> My component and Sandpack in a .astro page are here:
> https://github.com/sarah11918/sastro-v21/blob/main/src/components/experiments/MySandpack.jsx
> https://github.com/sarah11918/sastro-v21/blob/main/src/pages/experiments/codesandbox-sandpack.astro

**Reply by joeyfigaro** [reply]
Yes please! I've been banging my head on this today.

**Comment by Antonio476587** [comment]
How to use Astro in runtime for SSR without an adapter, like a renderToString function

**Reply by sarah11918** [reply]
✅ We have [the experimental Astro Container API page](https://docs.astro.build/en/reference/container-reference/) with `renderToString()` :partying_face:

**Comment by louiss0** [comment]
Well, once again I'm here to ask for the most commonly used Remark Plugins. #3037. Understanding the benefits of different remark plugins is difficult for beginners.

**Comment by teinett** [comment]
I receive mdx or html from API, for example, Wordpress post has custom WordpressBlock inside it's html. 
I have component to render this block in my Astro website: AstroComponentX
How can I connect my AstroComponentX and received html (set:html) or markdown (Markdown component) with WordpressBlock?

**Reply by delucis** [reply]
Sounds like a "working with remote content" recipe maybe? We do have a ["remote Markdown"](https://docs.astro.build/en/guides/markdown-content/#fetching-remote-markdown) section in docs already — maybe we pull that out to a recipe instead 🤔

**Reply by teinett** [reply]
I come back to this request for the recipe. 

I have a content-rich WordPress website. I want to keep using WordPress as CMS, but add AstroJS on the frontend. How can I do it? 
I don't understand how to receive and render WordPress Blocks (https://wordpress.org/blocks/) used in Posts and Pages.

**Comment by sarah11918** [comment]
How to deploy to cPanel

**Comment by sarah11918** [comment]
How to use the first paragraph of your blog text on a blog list/index page

**Comment by sarah11918** [comment]
How to make a lightbox where you can click on an optimized image and see the original full-resolution image?

**Reply by sarah11918** [reply]
Answer from Kapa:

You can achieve a lightbox functionality in Astro by using the `@julian_cataldo/astro-lightbox` component. This component provides a simple lightbox functionality that you can use in your Astro project.

Here's how you can use it:

1. Install the `@julian_cataldo/astro-lightbox` component. You can install it using npm, pnpm, or Yarn:

```bash
npm install @julian_cataldo/astro-lightbox
```
```bash
pnpm add @julian_cataldo/astro-lightbox
```
```bash
yarn add @julian_cataldo/astro-lightbox
```

2. Import the `@julian_cataldo/astro-lightbox` component into your Astro file:

```astro
---
import AstroLightbox from '@julian_cataldo/astro-lightbox';
---
```

3. Use the `AstroLightbox` component in your Astro file. You can pass the optimized image as the `src` prop and the original, high-definition image as the `data-src` prop:

```astro
<AstroLightbox src="/path/to/optimized/image.jpg" data-src="/path/to/original/image.jpg" />
```

In this example, clicking on the optimized image will open the original, high-definition image in a lightbox.

For more information, you can refer to the [`@julian_cataldo/astro-lightbox` component documentation](https://code.juliancataldo.com/component/astro-lightbox).

**Comment by sarah11918** [comment]
How do I create a content collection that only holds images?

**Reply by sarah11918** [reply]
Reply from Kapa:

You can create a simple content collection that only holds images in Astro by following these steps:

1. **Create a new collection**: Create a new directory inside `src/content/` for your images. For example, you could create a directory named `images`.

2. **Add images to the collection**: Add your image files to the new collection directory. Astro supports JPEG, PNG, SVG, and other common image file types.

3. **Define a collection schema**: Create a `src/content/config.ts` file to define a schema for your image collection. The schema can include an `image` helper to validate the image metadata using Zod.

Here's an example of how to define a schema for an image collection:

```javascript
import { defineCollection, z } from "astro:content";

const imageCollection = defineCollection({
  schema: ({ image }) => z.object({
    image: image().refine((img) => img.width >= 1080, {
      message: "Image must be at least 1080 pixels wide!",
    }),
  }),
});

export const collections = {
  images: imageCollection,
};
```
[source](https://docs.astro.build/en/guides/images/#images-in-content-collections)

In this example, the `image` helper is used to validate that the image is at least 1080 pixels wide. You can customize this validation to suit your needs.

4. **Query the collection**: Once you have a collection, you can start querying your images using Astro’s built-in content APIs. Here's an example of how to do this:

```astro
---
import { getCollection } from "astro:content";
const allImages = await getCollection("images");
---
{ allImages.map((image) => (
  <img src={image.data.image.src} alt={image.data.image.alt} />
))}
```
[source](https://docs.astro.build/en/guides/images/#images-in-content-collections)

In this example, the `getCollection` function is used to get all images from the `images` collection. The images are then rendered using the `<img>` element.

For more information, you can refer to the [Astro documentation](https://docs.astro.build/en/guides/images/#images-in-content-collections).

**Comment by sarah11918** [comment]
From Feedback Fish: How to do unit testing in Astro?

**Comment by sarah11918** [comment]
A more complete beginning-to-end example showing how middleware is used (anything, really) in recipe form

**Comment by sarah11918** [comment]
An example of using Astro DB for *something* practical, end to end.

**Reply by hfournier** [reply]
I'm looking at adding a comment section to my blog using Astro DB, maybe using server-islands... I could write it up, if that doesn't already exist.

**Reply by sarah11918** [reply]
Oooh, that sounds exciting, especially with server islands!

**Comment by sarah11918** [comment]
Guides for adding images from a digital asset management service like Contentful

**Comment by sarah11918** [comment]
The definitive build a carousel recipe!

**Comment by sarah11918** [comment]
add bootstrap

**Comment by sarah11918** [comment]
From Feedback Fish: I'm trying to implement a redirect for affiliate links on my website, I used the 'redirects' property in the config and it works perfectly. Though I'm getting this warning: [WARN] [redirects] Redirecting to an external URL is not officially supported..

Could you please let me know if there are other side effects of using external links in 'redirects' aside from the warning, and if this is not the correct way to do this, please let me know what is the best way to cloak and redirect affiliate links with Astro.

**Comment by sarah11918** [comment]
Construct a canonical URL from `Astro.url`

(and then finally remove the API reference entry to `Astro.canonicalURL` that was deprecated in v1 and removed in v2.

**Comment by sarah11918** [comment]
How to carry event listeners through page transitions

**Comment by sarah11918** [comment]
Combining server islands and interactive islands

**Comment by sarah11918** [comment]
How to activate a React component when a button is clicked.

Kapa AI spit out something like this:

```
import { useState } from 'react';

export default function MyComponent() {
  const [isActive, setIsActive] = useState(false);

  const handleClick = () => {
    setIsActive(!isActive);
  };

  return (
    <div>
      <a href="#" onClick={handleClick}>Activate Component</a>
      {isActive && <YourReactComponent />}
    </div>
  );
}
```

**Comment by sarah11918** [comment]
3 level dynamic routes optimization: My dynamic routes look like this /[type]/[provider]/[slug] and every level reads the same json file, how can I read json file once and generate all these pages (currently it reads every level)

Kapa AI spit out something like this:

```
---
import Product from '../components/Product.astro';
import Store from '../components/Store.astro';
import Handbook from '../components/Handbook.astro';

const { slug } = Astro.request.params;
const { title, text } = Astro.props;
---

<html>  
  <head>    
    <title>{title}</title>  
  </head>  
  <body>    
    {slug.length === 0 && <Store title={title} text={text} />}
    {slug.length === 1 && slug[0] === 'products' && <Product title={title} text={text} />}
    {slug.length === 2 && slug[0] === 'products' && slug[1] === 'astro-handbook' && <Handbook title={title} text={text} />}
  </body>
</html>
```

> In this example, Store, Product, and Handbook are Astro components that represent different UIs for different levels of your route. You can replace these with your own components.
Remember to import the components at the top of your Astro file. The slug parameter is an array that contains the parts of your route. You can check the length of the slug array to determine the depth of your route and render different components accordingly.

**Comment by sarah11918** [comment]
How to set a global variable, sitewide.  I want to set my website name as a variable, so I can refer to it as `{brand}` instead of typing it out every time. How do I do this?

**Comment by sarah11918** [comment]
Is there a way to render Jupyter notebooks with Astro? I have a .ipynb file, and I'd like to render it to HTML.

**Comment by sarah11918** [comment]
Optimal Filter + List I have a list of items that I want to make filterable by putting an interactive filter above. How would I do this optimally? Generally I'd like to SSG this page and sent as little javascript to the client as possible

**Comment by sarah11918** [comment]
An example of using Astro.cookies in combination with manipulating the cookies from a framework (e.g. React) component.

**Comment by sarah11918** [comment]
Creating a global toggle I'll create a button for this website labeled "Zen Mode". When this button is clicked, I want to display an alternate rendering for the site, across all pages. I also want to store this state in a cookie. How should I approach this?

Kapa spit out:

```
<button id="zenModeToggle">Zen Mode</button>

<style>
  /* Add your CSS for Zen Mode and normal mode here */
</style>

<script is:inline>
  const mode = (() => {
    if (typeof localStorage !== 'undefined' && localStorage.getItem('mode')) {
      return localStorage.getItem('mode');
    }
    return 'normal';
  })();

  if (mode === 'normal') {
    document.documentElement.classList.remove('zen');
  } else {
    document.documentElement.classList.add('zen');
  }

  window.localStorage.setItem('mode', mode);

  const handleToggleClick = () => {
    const element = document.documentElement;
    element.classList.toggle("zen");

    const isZen = element.classList.contains("zen");
    localStorage.setItem("mode", isZen ? "zen" : "normal");

    // Set a cookie for the current mode
    Astro.cookies.set("mode", isZen ? "zen" : "normal");
  }

  document.getElementById("zenModeToggle").addEventListener("click", handleToggleClick);
</script>
```

**Reply by delucis** [reply]
A community member built a Starlight plugin to add a feature like this to Starlight sites: https://github.com/trueberryless/starlight-view-modes

**Comment by sarah11918** [comment]
import images as blob. Making an endpoint, and would like to return a Response.  I have a folder with a list of icons. Is it possible to `import.meta.glob` these images somehow so that I can return an image from the list as a blob in my endpoint?

**Comment by sarah11918** [comment]
Global Fetch Headers How do I apply a bearer token to all my fetch requests globally?

**Comment by sarah11918** [comment]
Cookie button How can I make a button that sets a cookie?

Kapa AI spit out:

```
<button id="setCookie">Set Cookie</button>
```

```
<script>
  const button = document.querySelector('#setCookie');
  button.addEventListener('click', () => {
    Astro.cookies.set('key', 'value');
  });
</script>
```

In this example, when the button with the id setCookie is clicked, a cookie with the key `key` and the value `value` is set.
Remember that `Astro.cookies.set()` method can also accept an options object as a third argument, which allows you to customize how the cookie is serialized. For example, you can set the `maxAge` or `httpOnly` options among others. You can find more details about these options in the [Astro documentation](https://docs.astro.build/en/reference/api-reference/#astrocookiesetoptions).
Please note that this code should be executed in a server-side context, as `Astro.cookies` is only available in server-side rendering mode.

**Comment by sarah11918** [comment]
Seed database I have a csv file that I would like to seed my Astro DB with. How to do this?

**Comment by sarah11918** [comment]
Add data to nanostores with form action Hello, how can I add data to a nanostores atom from my frontmatter using form actions?

Discord ai support thread here: https://discord.com/channels/830184174198718474/1248682440512241718

**Reply by Fryuni** [reply]
Going through some old recipe ideas. This one can't be done easily with just Astro and Nanostores because frontmatter runs on the server and pure Nanostores should be used only on the client. They assume the isolation of the client so everything is globally shared, doing so on the server would leak state between users.

It _can_ be done if you make Nanostores request-aware and to send data to the client, but that requires some extra magic. One way to do it is through ALS and a state-script, as implemented in my `@inox-tools/request-nanostores` (which could be used in a recipe).
Another way would be to create a `<script>` in the component that sets the value on the client using something from the frontmatter, it is less magical but also less flexible and more verbose.

**Comment by sarah11918** [comment]
Using SVG icons with the `<Image>` component and/or in Markdown

e.g. How can I use an SVG icon inside an Image object? Say I have a collection of markdown files, where on the frontmatter I place the SVG image's path. When creating the object which will render each element of the collection, I want to use the <Image> object with the SVG image path, is this possible?

**Comment by sarah11918** [comment]
Deploy vercel: A function of my application downloads files from an S3 bucket and stores them in a temporary folder. It works locally but in production it gives me an error. Is there a possibility of creating this folder in production?

**Comment by sarah11918** [comment]
Starlight supports a built-in theme switcher for users to decide using light or dark mode. If I have a picture that requires to be displayed differently in light/dark modes, how do I achieve this?

From Chris:
There is although it’s not fantastic, basically include the two variants and then use the light:sl-hidden class name on the dark one, and dark:sl-hidden on the light one.
We use this internally in a couple of places because unfortunately colour scheme support in <picture> doesn’t work in Safari (does in all the other browsers…)

**Comment by sarah11918** [comment]
[recipe for analyzing bundle size](https://github.com/withastro/docs/issues/9164)
 
 From the linked issue:
 
 When using client-side JavaScript libraries, it is important to keep track of how much they will increase the final js bundle size. Knowing this, we can optimize our page performance by potentially removing, or lazy-loading our client-side code. An official guide on how to analyze/visualize our page's js budle would be super helpful to do this. 

For reference, some other frameworks have guides on how to do this:
- https://nextjs.org/docs/pages/building-your-application/optimizing/package-bundling#analyzing-javascript-bundles
- https://nuxt.com/docs/api/nuxt-config#analyze
- https://remix.run/docs/en/main/guides/performance#bundle-analysis

**Comment by Newman5** [comment]
Heads up - 
the link is broken:  [Check out our full guide to writing recipes.](https://github.com/withastro/docs/blob/main/contributor-guides/submitting-a-recipe.md)

might I suggest: [Check out our full guide to writing recipes.](https://contribute.docs.astro.build/guides/recipe-writing/)

Happy Showcase Friday! Vive le Docs!

**Reply by sarah11918** [reply]
Thanks @Newman5 ! Ignore any earlier comment of mine where I thought you were trying to tell me a DOCS link was broken! 😅 I'll update the top post in this discussion. 🚀

**Comment by Newman5** [comment]
Heads up - 
the link is broken:  [Check out our full guide to writing recipes.](https://github.com/withastro/docs/blob/main/contributor-guides/submitting-a-recipe.md)

might I suggest: [Check out our full guide to writing recipes.](https://contribute.docs.astro.build/guides/recipe-writing/)

Happy Showcase Friday! Vive le Docs!

**Reply by sarah11918** [reply]
(I think this is a duplicate that showed up again somehow? The link appears fixed to me.)

**Reply by Newman5** [reply]
Yep.  It's working - thank you!
 I can delete this comment or you can.  thanks for getting it fixed.
